src/lib/Libwin/util.c

```diff
@@ -978,3 +978,61 @@ revert_impersonated_user()
 	}
 	return result;
 }
+
+/* @brief
+ *  	A wrapper to CloseHandle()
+ *  	If CloseHandle fails, log the error.
+ *
+ * @param[in]   hObject : handle to be closed
+ * @param[in]   handlename : name of param containing handle
+ * @param[in]   funcname : function name where API was invoked
+ * @param[in]   lineno : line number of API invocation
+ *
+ * @return     BOOL
+ * @retval     whether CloseHandle succeeded, return value of CloseHandle() API
+ */
+BOOL
+_log_wrap_CloseHandle(HANDLE hObject, LPCSTR handlename, LPCSTR funcname, INT lineno)
+{
+	BOOL rc;
+
+#undef CloseHandle
+	rc = CloseHandle(hObject);
+#define CloseHandle(param) _log_wrap_CloseHandle(param, #param, __func__, __LINE__)
+
+	if (!rc) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_ERR, funcname,
+			"CloseHandle(%s) at line %d failed with error %d, Handle[%x]",
+			handlename, lineno, GetLastError(), hObject);
+	}
+	return rc;
+}
+
+/* @brief
+ *  	A wrapper to LocalFree()
+ *  	If LocalFree fails, log the error.
+ *
+ * @param[in]   hObject : handle to be closed
+ * @param[in]   handlename : name of param containing handle
+ * @param[in]   funcname : function name where API was invoked
+ * @param[in]   lineno : line number of API invocation
+ *
+ * @return     BOOL
+ * @retval     whether LocalFree succeeded, return value of LocalFree() API
+ */
+HLOCAL
+_log_wrap_LocalFree(HLOCAL hObject, LPCSTR handlename, LPCSTR funcname, INT lineno)
+{
+	HLOCAL hret;
+
+#undef LocalFree
+	hret = LocalFree(hObject);
+#define LocalFree(param) _log_wrap_LocalFree(param, #param, __func__, __LINE__)
+
+	if (hObject && (hObject == hret)) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_ERR, funcname,
```

Comment:
same here add the server event class

Reply:
https://github.com/PBSPro/pbspro/pull/1727/commits/d40ea6a01f1c873f0de3a7e67671b33fec9927cc